src/common/com/intellij/plugins/haxe/lang/psi/HaxeResolver.java

```diff
@@ -250,7 +250,13 @@ private PsiFile resolveImportFile(HaxeReference reference) {
         VirtualFile importVFile = importDir == null ? null : importDir.findChild(fileName);
         importPsiFile = importVFile == null ? null : PsiManager.getInstance(reference.getProject()).findFile(importVFile);
         if (importPsiFile != null) {
-          break;
+          // for addition case-sensetive check because find file is not case-sensetive
+          if(!fileName.equals(importPsiFile.getName())) {
```

Comment:
Did you test this on Windows?  With Windows not having case sensitive file names, I wonder whether the filename returned from FindFile is really going to match the case of the import file name, and whether the case of the file names on a typical haxe SDK installation is all caps.